flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/functions/casting/rules/MapToStringCastRule.java

```diff
@@ -0,0 +1,227 @@
+                                                                    "append",
+                                                                    strLiteral(", "))))
+                                    // Declare key and values variables
+                                    .declPrimitiveStmt(keyType, keyTerm)
+                                    .declStmt(
+                                            boolean.class,
+                                            keyIsNullTerm,
+                                            methodCall(keyArrayTerm, "isNullAt", indexTerm))
+                                    .declPrimitiveStmt(valueType, valueTerm)
+                                    .declStmt(
+                                            boolean.class,
+                                            valueIsNullTerm,
+                                            methodCall(valueArrayTerm, "isNullAt", indexTerm))
+                                    // Execute casting if inner key/value not null
+                                    .ifStmt(
+                                            "!" + keyIsNullTerm,
+                                            thenBodyWriter ->
+                                                    thenBodyWriter
+                                                            // If key not null, extract it and
+                                                            // execute the cast
+                                                            .assignStmt(
+                                                                    keyTerm,
+                                                                    rowFieldReadAccess(
+                                                                            indexTerm,
+                                                                            keyArrayTerm,
+                                                                            keyType))
+                                                            .append(keyCast)
+                                                            .stmt(
+                                                                    methodCall(
+                                                                            builderTerm,
+                                                                            "append",
+                                                                            keyCast
+                                                                                    .getReturnTerm())),
+                                            elseBodyWriter ->
+                                                    elseBodyWriter.stmt(
+                                                            methodCall(
+                                                                    builderTerm,
+                                                                    "append",
+                                                                    strLiteral("NULL"))))
```

Comment:
Could you please extract all usages `NULL`, `<NULL>`, etc. to constants in the superclass?

Reply:
This PR doesn't deal with `<NULL>`, because that's user facing (see `PrintUtils#NULL_COLUMN`), the next PR (for FLINK-24685) will deal with it.

For `NULL`, I've added a constant in `PrintUtils`

Reply:
thx